Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -1103,6 +1034,87 @@ void SourceBufferPrivate::append(Vector<unsigned char>&&)
     RELEASE_ASSERT_NOT_REACHED();
 }
 
+void SourceBufferPrivate::memoryPressure(uint64_t maximumBufferSize, const MediaTime& currentTime, bool isEnded, CompletionHandler<void(bool)>&& completionHandler)
+{
+    auto oldBuffered = m_buffered->ranges();
+    evictFrames(maximumBufferSize, maximumBufferSize, currentTime, isEnded);
+    completionHandler(m_buffered->ranges() != oldBuffered);
+}
+
+bool SourceBufferPrivate::evictFrames(uint64_t newDataSize, uint64_t maximumBufferSize, const MediaTime& currentTime, bool isEnded)
+{
+    bool isBufferFull = true;
+
+    const auto& buffered = m_buffered->ranges();
+
+    // FIXME: All this is nice but we should take into account negative playback rate and begin from after current time
+    // and be more conservative with before current time.
+
+    unsigned timeChunkAsMilliseconds = evictionAlgorithmInitialTimeChunk;
+    do {
+        const MediaTime timeChunk = MediaTime(timeChunkAsMilliseconds, 1000);
+        const MediaTime maximumRangeEnd = std::min(currentTime - timeChunk, findPreviousSyncSamplePresentationTime(currentTime));
+
+        do {
+            MediaTime rangeStart = buffered.minimumBufferedTime();
+            MediaTime rangeEnd = std::min(rangeStart + timeChunk, maximumRangeEnd);
+
+            if (rangeStart >= rangeEnd)
+                break;
+
+            // 4. For each range in removal ranges, run the coded frame removal algorithm with start and
+            // end equal to the removal range start and end timestamp respectively.
+            removeCodedFrames(rangeStart, rangeEnd, currentTime, isEnded);
+            MediaTime newRangeStart = buffered.minimumBufferedTime();
+            if (newRangeStart == rangeStart)
+                break;
```

Comment:
Nit: don't really need this as a local variable.

Reply:
that's existing code I moved; though I can see the value during development/debugging. Easier to step in the debugger and look at the value of the variable after a breakpoint